core/base-service/coalesce-badge.spec.js

```diff
@@ -0,0 +1,257 @@
+'use strict'
+
+const { expect } = require('chai')
+const { getShieldsIcon } = require('../../lib/logos')
+const coalesceBadge = require('./coalesce-badge')
+
+describe('coalesceBadge', function() {
+  describe('Label', function() {
+    it('uses the default label', function() {
+      expect(coalesceBadge({}, {}, { label: 'heyo' }).text).to.deep.equal([
+        'heyo',
+        'n/a',
+      ])
+    })
+
+    // This behavior isn't great and we might want to remove it.
+    it('uses the category as a default label', function() {
+      expect(coalesceBadge({}, {}, {}, { category: 'cat' }).text).to.deep.equal(
+        ['cat', 'n/a']
+      )
+    })
+
+    it('preserves an empty label', function() {
+      expect(
+        coalesceBadge({}, { label: '', message: '10k' }, {}).text
+      ).to.deep.equal(['', '10k'])
+    })
+
+    it('overrides the label', function() {
+      expect(
+        coalesceBadge({ label: 'purr count' }, { label: 'purrs' }, {}).text
+      ).to.deep.equal(['purr count', 'n/a'])
+    })
+  })
+
+  describe('Message', function() {
+    it('applies the service message', function() {
+      expect(coalesceBadge({}, { message: '10k' }, {}).text).to.deep.equal([
+        undefined,
+        '10k',
+      ])
+    })
+
+    it('applies a numeric service message', function() {
+      // While a number of badges use this, in the long run we may want
+      // `render()` to always return a string.
+      expect(coalesceBadge({}, { message: 10 }, {}).text).to.deep.equal([
+        undefined,
+        10,
+      ])
+    })
+  })
+
+  describe('Right color', function() {
+    it('uses the default color', function() {
+      expect(coalesceBadge({}, {}, {}).color).to.equal('lightgrey')
+    })
+
+    it('overrides the color', function() {
+      expect(
+        coalesceBadge({ colorB: '10ADED' }, { color: 'red' }, {}).color
+      ).to.equal('10ADED')
+    })
+
+    context('In case of an error', function() {
+      it('does not override the color', function() {
+        expect(
+          coalesceBadge(
+            { colorB: '10ADED' },
+            { isError: true, color: 'lightgray' },
+            {}
+          ).color
+        ).to.equal('lightgray')
+      })
+    })
+
+    it('applies the service color', function() {
+      expect(coalesceBadge({}, { color: 'red' }, {}).color).to.equal('red')
+    })
+  })
+
+  describe('Left color', function() {
+    it('provides no default label color', function() {
+      expect(coalesceBadge({}, {}, {}).labelColor).to.be.undefined
+    })
+
+    it('applies the service label color', function() {
+      expect(coalesceBadge({}, { labelColor: 'red' }, {}).labelColor).to.equal(
+        'red'
+      )
+    })
+
+    it('overrides the label color', function() {
+      expect(
+        coalesceBadge({ colorA: '42f483' }, { color: 'green' }, {}).labelColor
```

Comment:
Just trying to get my bearings around `colorA` vs `labelColor` and `colorB` vs `color`. Are the two on the left (A & B) still valid, but deprecated aliases for the labelColor and color or are A & B still undeprecated/promoted?

Reply:
As of now, still undeprecated.

Reply:
Gotcha, thanks